lib/archethic/oracle_chain/services/uco_price/providers/coin_paprika.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Archethic.OracleChain.Services.UCOPrice.Providers.CoinPaprika do
+  @moduledoc false
+
+  alias Archethic.OracleChain.Services.UCOPrice.Providers.Impl
+
+  @behaviour Impl
+
+  require Logger
+
+  @impl Impl
+  @spec fetch(list(binary())) :: {:ok, %{required(String.t()) => any()}} | {:error, any()}
+  def fetch(pairs) when is_list(pairs) do
+    pairs_str = Enum.join(pairs, ",")
+
+    httpc_options = [
+      ssl: [
+        verify: :verify_peer,
+        cacertfile: CAStore.file_path(),
+        depth: 2,
+        customize_hostname_check: [
+          match_fun: :public_key.pkix_verify_hostname_match_fun(:https)
+        ]
+      ],
+      connect_timeout: 1000,
+      timeout: 2000
+    ]
+
+    query =
+      String.to_charlist(
+        "https://api.coinpaprika.com/v1/coins/uco-uniris/markets?quotes=#{pairs_str}"
+      )
+
+    with {:ok, {{_, 200, 'OK'}, _headers, body}} <-
+           :httpc.request(:get, {query, []}, httpc_options, []),
+         {:ok, payload} <- Jason.decode(body) do
+      quotes =
+        payload
+        |> Enum.map(fn %{"quotes" => quotes} -> quotes end)
+
+      prices =
+        pairs
+        |> Enum.map(fn pair ->
+          {pair, Enum.map(quotes, &get_in(&1, [String.upcase(pair), "price"]))}
```

Comment:
Here there might be a problem. The `get_in` function returns `nil` if no value is found. But we cannot have a nil value in the list of prices, because aggregator cannot handle it. We should ensure there is not nil value in list of prices.

Reply:
what should be the default value in this case ?

Reply:
Yes, if there is a nil value, just remove it from the list and if there is only nil value, return an error or an empty list